test-integ/catalogv2/explicit_destinations_l7_test.go

```diff
@@ -0,0 +1,486 @@
+
+	sp := sprawltest.Launch(t, cfg)
+
+	var (
+		asserter = topoutil.NewAsserter(sp)
+
+		topo    = sp.Topology()
+		cluster = topo.Clusters["dc1"]
+
+		ships = topo.ComputeRelationships()
+	)
+
+	clientV2 := sp.ResourceServiceClientForCluster(cluster.Name)
+
+	t.Log(topology.RenderRelationships(ships))
+
+	// Make sure things are in v2.
+	libassert.CatalogV2ServiceHasEndpointCount(t, clientV2, "static-client", nil, 1)
+	libassert.CatalogV2ServiceHasEndpointCount(t, clientV2, "static-server-v1", nil, 1)
+	libassert.CatalogV2ServiceHasEndpointCount(t, clientV2, "static-server-v2", nil, 1)
+	libassert.CatalogV2ServiceHasEndpointCount(t, clientV2, "static-server", nil, 0)
+
+	// Check relationships
+	for _, ship := range ships {
+		t.Run("relationship: "+ship.String(), func(t *testing.T) {
+			var (
+				svc = ship.Caller
+				u   = ship.Upstream
+			)
+
+			v1ID := ship.Upstream.ID
+			v1ID.Name = "static-server-v1"
+			v1ClusterPrefix := clusterPrefix(u.PortName, v1ID, u.Cluster)
+
+			v2ID := ship.Upstream.ID
+			v2ID.Name = "static-server-v2"
+			v2ClusterPrefix := clusterPrefix(u.PortName, v2ID, u.Cluster)
+
+			switch u.PortName {
+			case "tcp":
+				// we expect 2 clusters, one for each leg of the split
+				asserter.UpstreamEndpointStatus(t, svc, v1ClusterPrefix+".", "HEALTHY", 1)
+				asserter.UpstreamEndpointStatus(t, svc, v2ClusterPrefix+".", "HEALTHY", 1)
+
+				// Both should be possible.
+				v1Expect := fmt.Sprintf("%s::%s", cluster.Name, v1ID.String())
+				v2Expect := fmt.Sprintf("%s::%s", cluster.Name, v2ID.String())
+
+				asserter.CheckBlankspaceNameTrafficSplitViaTCP(t, svc, u,
+					map[string]int{v1Expect: 10, v2Expect: 90}, 2)
+
+			case "grpc":
+				// we expect 2 clusters, one for each leg of the split
+				asserter.UpstreamEndpointStatus(t, svc, v1ClusterPrefix+".", "HEALTHY", 1)
+				asserter.UpstreamEndpointStatus(t, svc, v2ClusterPrefix+".", "HEALTHY", 1)
+
+				// Both should be possible.
+				v1Expect := fmt.Sprintf("%s::%s", cluster.Name, v1ID.String())
+				v2Expect := fmt.Sprintf("%s::%s", cluster.Name, v2ID.String())
+
+				asserter.CheckBlankspaceNameTrafficSplitViaGRPC(t, svc, u,
+					map[string]int{v1Expect: 10, v2Expect: 90}, 2)
+
+			case "http":
+				// we expect 2 clusters, one for each leg of the split
+				asserter.UpstreamEndpointStatus(t, svc, v1ClusterPrefix+".", "HEALTHY", 1)
+				asserter.UpstreamEndpointStatus(t, svc, v2ClusterPrefix+".", "HEALTHY", 1)
+
+				// Both should be possible.
+				v1Expect := fmt.Sprintf("%s::%s", cluster.Name, v1ID.String())
+				v2Expect := fmt.Sprintf("%s::%s", cluster.Name, v2ID.String())
+
+				asserter.CheckBlankspaceNameTrafficSplitViaHTTP(t, svc, u, false, "/",
+					map[string]int{v1Expect: 10, v2Expect: 90}, 2)
+			case "http2":
+				asserter.UpstreamEndpointStatus(t, svc, v1ClusterPrefix+".", "HEALTHY", 1)
+
+				// Only v1 is possible.
+				asserter.CheckBlankspaceNameViaHTTP(t, svc, u, true, "/", cluster.Name, v1ID)
```

Comment:
Legit bug: the http1/2 requests here were using the nginx forward proxy by setting the `Proxy` field on the `http.Transport`. For this specific test case (http2) I was forcing `h2c` which had the side effect of clearing the `Proxy` field because Go doesn't support `http2` + proxies.

On linux (my machine and CI) the proxy is irrelevant because there's no intervening VM and the un-proxied requests worked fine. On macs the VM prevents using the real addresses, so I switched this to use the exposed docker ports instead when http2 is in use.